lib/salus/scanners/yarn_audit.rb

```diff
@@ -128,47 +126,172 @@ def handle_legacy_yarn_audit
       chdir = File.expand_path(@repository&.path_to_repo)
 
       Salus::YarnLock.new(File.join(chdir, 'yarn.lock')).add_line_number(vulns)
+
+      run_auto_fix(generate_fix_feed)
+
       vulns = combine_vulns(vulns)
       log(format_vulns(vulns))
       report_stdout(vulns.to_json)
       report_failure
     end
 
-    def auto_fix_vulns
-      @vulns_w_paths.each do |vuln|
-        path = vuln["Path"]
-        package = vuln["Package"]
-        # vuln["Path"] for some indirect dependency "foo"
-        # looks like "bar > baz > foo".
-        # vuln["Path"] for some direct dependency "foo"
-        # looks like "foo"
-        is_direct_dep = path == package
-
-        # TODO: Fix indirect dependencies as well
-        fix_direct_dependency(vuln) if is_direct_dep
+    def generate_fix_feed
+      actions = []
+      grouped_vulns = @vulns_w_paths.group_by { |h| [h["Package"], h["Patched in"]] }
+      grouped_vulns.each do |key, values|
+        name = key.first
+        patch = key.last
+        resolves = []
+        values.each do |value|
+          resolves.append({
+                            "id": value["ID"],
+                  "path": value["Path"],
+                  "dev": false,
+                    "optional": false,
+                    "bundled": false
+                          })
+        end
+        actions.append({
+                         "action": "update",
+          "module": name,
+          "target": patch,
+          "resolves": resolves
+                       })
       end
+      actions
+    end
+
+    def run_auto_fix(feed)
+      fix_indirect_dependency(feed)
+      fix_direct_dependency(feed)
     rescue StandardError
       report_error("An error occurred while auto-fixing vulnerabilities")
     end
 
-    def fix_direct_dependency(vuln)
-      package = vuln["Package"]
-      patched_version_range = vuln["Patched in"]
+    def fix_direct_dependency(feed)
+      @packages = JSON.parse(@repository.package_json)
+      feed.each do |vuln|
+        patch = vuln[:target]
+        resolves = vuln[:resolves]
+        package = vuln[:module]
+        resolves.each do |resolve|
+          if patch != "No patch available" && package == resolve[:path]
```

Comment:
is it possible for `patch` to be `nil` or empty string?

Reply:
I haven't seen an example of it but i can add a check